### DISCUSSION TITLE
Testing with transactions

### DISCUSSION THREAD

**Discussion by bluemoon** [SeaQL/sea-orm#596] [Q&A]
I'm using axum as my web server and I would like to write tests that use a transaction (I cannot use mock because it is not `Clone`). axum requires that `Extension` is `Clone`. I'm wondering what the best way to deal with this is? 

I assume `DatabaseTransaction` is also not clone because of the same issue in #438

**Comment by billy1624** [comment]
Hey @bluemoon, sorry for the delay. This problem bother me for quite some time, I think we need to make our mock utility cloneable. I'll make a PR later.

Btw... do you have any updates on this? How you tackle it?

**Reply by bluemoon** [reply]
I forgot about this, I actually have just avoided testing because I thought it would be too messy currently

**Reply by billy1624** [reply]
Alright, great!
